### ISSUE TITLE
Input type="date" polyfill (date picker): arrow keys do not work as expected in the calendar grid of days  and confusing focus issues 

### ISSUE THREAD

**Issue by dfarough** [wet-boew/wet-boew#7410]
@wet-boew/access-working-group 
Using Jaws with Internet explorer 11 and Firefox, the arrow keys do not work as per ARIA datepicker keyboard specifications.  As a result the visible focus does not change when using arrow keys to move from cell to cell within the calendar grid.  IF a user is using a screen magnification program as well as a screen reader program like Jaws, the visible focus will not change when moving from cell to cell within the calendar grid. but Jaws may speak the value of the next cell.  this would be confusing.

The only way a user can correct this behaviour at present is to disable the virtual cursor or browse mode by first pressing Insert plus z.  Some users may not be aware of this remedy.

in issue #6670 the possibility of using the application role for the date pickerwas offered as a solution for this issue, but I am not aware whether any testing of this was done using Jaws and NVDA.

This is still a problem using the date picker widget in WET 4.

**Comment by LaurentGoderre** [comment]
Thanks. Maybe I can create a test scenario that uses `role="application"` before comitting to this approach. Sounds good?

**Comment by LaurentGoderre** [comment]
@dfarough I create a demo with the `role=application`. Can you test and let me know?

http://laurentgoderre.github.io/wet-boew/unmin/demos/datepicker/datepicker-en.html

**Comment by dfarough** [comment]
@LaurentGoderre This is working better.  the arrow keys are working as expected.  If I go to the datepicker for the end date, I cannot specify a different  month in the select.  the values do not appear to be greyed out so I am wondering why this does not behave as expected.

I should be able to expand the list by using alt + down arrow, this works, but pressing down or up arrow will not allow me to select a different month.

I will test this using NVDA and report on that separately.  Browser in use at present is Internet explorer 11.  I'll check with firefox and report on that as well.

**Comment by dfarough** [comment]
@LaurentGoderre 
The behaviour I mentioned earlier with regard to selecting values from the drop downs for month and year does not seem to be present using Firefox.  I have however noticed the following issue.

If you open the month select for the datepicker when specifying an end date. If you press alt + down arrow to open the month select and arrow to a value, pressing tab at this point should move you to the calendar grid.  this does not happen.  Instead the focus moves out of the date picker and back to the end date field.

Another issue I am noticing is that if I move to the calendar grid and arrow to a new date, and then press shift + tab to move back to the controls at the top of the dialogue, my focus moves back to the previous date that had been specified in the calendar grid.

I am wondering if it might be better if the whole dialogue used the application role.

I have not noticed any adverse effects using NVDA with this dialogue.  The behaviour I mentioned earlier with the select dropdowns occurs with NVDA and Internet explorer 11 as well.  I wonder if this occurs when a mixture of valid and invalid values occur within a dropdown in IE11.

**Comment by LaurentGoderre** [comment]
Would applying the `role=application` to the whole calendar interfere with the valid form controls? If you don't think it will, I can create a new demo.

I would also prefer to focus on one issue at the time because otherwise, it is very difficult to manage. We can deal with the other issues once we settle on the use of `role=application`

**Comment by dfarough** [comment]
This should not effect the form controls.  The AT should speak the labels as appropriate.  there is no non focusable text in this dialogue so I don't see a problem with this approach.  It might even fix some of the tabbing issues I mentioned earlier.

**Comment by LaurentGoderre** [comment]
Ok, try it again http://laurentgoderre.github.io/wet-boew/unmin/demos/datepicker/datepicker-en.html

**Comment by dfarough** [comment]
THE APPLICATION ROLE WORKS WELL WITH BOTH Jaws  and nvda.
Would you like me to open separate issues for the tab order for this dialogue?

**Comment by LaurentGoderre** [comment]
I will merge the `role=application` fix and close this issue, Please create a new issue for the next thing we tackle.
